both/methods.js

```diff
@@ -153,16 +153,25 @@ Meteor.methods({
 
     Meteor.call('refreshRoundCodes', competitionId, eventCode);
   },
-  addNonEventRound: function(competitionId, round) {
+  addScheduleEvent: function(competitionId, eventData, roundId) {
     check(competitionId, String);
     throwIfCannotManageCompetition(this.userId, competitionId);
-    return Rounds.insert({
+
+    var round = Rounds.findOne(roundId);
+
+    return ScheduleEvents.insert({
       competitionId: competitionId,
-      title: round.title,
-      startMinutes: round.startMinutes,
-      durationMinutes: round.durationMinutes,
+      roundId: (round ? round._id : null),
```

Comment:
So there are 2 things here, neither of which I'm too familiar with:

1. Methods that are defined both client and server side. As part of so called "latency compensation", client side js code that calls a method will immediately invoke the method client side (this is called a "stub"). I do not know what happens if that client side method invokation throws an exception. I would imagine that it is ignored.
2. Methods that are defined server side can throw exceptions. According to [this guy](http://stackoverflow.com/a/20154120), those exceptions show up as the err argument of the optional callback to to `Meteor.call`. Is that the "error handling callback" you mentioned? If so, I'm confused by the word "thread" when talking about client side js code.

What's an example of a method we have that throws an exception and doesn't do what's intended?